tests/unit/domain/test_chat_context.py

```
@@ -0,0 +1,454 @@
+        super().__init__()
+        self.added_messages = []
+        self.updated_messages = []
+        self.removed_messages = []
+        self.expired_removed = False
+
+    def add_message(self, message, message_collection, limits):
+        self.added_messages.append(message)
+        messages = list(message_collection.messages.values())
+        messages.append(message)
+        if limits and limits.max_messages is not None:
+            messages = messages[-limits.max_messages :]
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)